TidepoolKit/Model/Datum/TCGMSettingsDatum.swift

```diff
@@ -0,0 +1,240 @@
+        try container.encodeIfPresent(model, forKey: .model)
+        try container.encodeIfPresent(serialNumber, forKey: .serialNumber)
+        try container.encodeIfPresent(transmitterId, forKey: .transmitterId)
+        try container.encodeIfPresent(units, forKey: .units)
+        try container.encodeIfPresent(defaultAlerts, forKey: .defaultAlerts)
+        try container.encodeIfPresent(scheduledAlerts, forKey: .scheduledAlerts)
+        try container.encodeIfPresent(highAlertsDEPRECATED, forKey: .highAlertsDEPRECATED)
+        try container.encodeIfPresent(lowAlertsDEPRECATED, forKey: .lowAlertsDEPRECATED)
+        try container.encodeIfPresent(outOfRangeAlertsDEPRECATED, forKey: .outOfRangeAlertsDEPRECATED)
+        try container.encodeIfPresent(rateOfChangeAlertsDEPRECATED, forKey: .rateOfChangeAlertsDEPRECATED)
+        try super.encode(to: encoder)
+    }
+
+    public struct Alerts: Codable, Equatable {
+        public var enabled: Bool?
+        public var urgentLow: LevelAlert?
+        public var urgentLowPredicted: LevelAlert?
+        public var low: LevelAlert?
+        public var lowPredicted: LevelAlert?
+        public var high: LevelAlert?
+        public var highPredicted: LevelAlert?
+        public var fall: RateAlert?
+        public var rise: RateAlert?
+        public var noData: DurationAlert?
+        public var outOfRange: DurationAlert?
+
+        public init(enabled: Bool, urgentLow: LevelAlert? = nil, urgentLowPredicted: LevelAlert? = nil, low: LevelAlert? = nil, lowPredicted: LevelAlert? = nil, high: LevelAlert? = nil, highPredicted: LevelAlert? = nil, fall: RateAlert? = nil, rise: RateAlert? = nil, noData: DurationAlert? = nil, outOfRange: DurationAlert? = nil) {
+            self.enabled = enabled
+            self.urgentLow = urgentLow
+            self.urgentLowPredicted = urgentLowPredicted
+            self.low = low
+            self.lowPredicted = lowPredicted
+            self.high = high
+            self.highPredicted = highPredicted
+            self.fall = fall
+            self.rise = rise
+            self.noData = noData
+            self.outOfRange = outOfRange
+        }
+
+        public struct DurationAlert: Codable, Equatable {
+            public enum Units: String, Codable {
+                case hours
+                case minutes
+                case seconds
+            }
+
+            public var enabled: Bool?
+            public var duration: Double?
+            public var units: Units?
+            public var snooze: Snooze?
+
+            public init(enabled: Bool, duration: Double, units: Units, snooze: Snooze? = nil) {
+                self.enabled = enabled
+                self.duration = duration
+                self.units = units
+                self.snooze = snooze
+            }
+        }
+
+        public struct LevelAlert: Codable, Equatable {
+            public typealias Units = TBloodGlucose.Units
```

Comment:
Should these units be allowed to be different than the CGM settings units?

Reply:
The backend is doing all validation enforcement. Much like documenting, I really don't want to get into the business of having to keep the validations in sync between this code and the backend.

I suppose I could remove these units, but this is reflecting an actually used and deployed API, not anything brand new. So, going to leave this as-is.